src/file-manager/fm-directory-view.c

```diff
@@ -8743,7 +8743,14 @@ real_update_menus (FMDirectoryView *view)
 	for (l = selection; l != NULL; l = l->next) {
 		CajaFile *file;
 
-		file = CAJA_FILE (selection->data);
+		file = CAJA_FILE (l->data);
```

Comment:
I threw this change in because the code looked wrong. I think this was a bug, but it was being made harmless by the behavior of caja_mime_get_default_application_for_files-- see the commit message for details.